client/src/transmitter/cast/websocket.ts

```diff
@@ -0,0 +1,158 @@
+import isIp from 'is-ip'
+
+const isLocalHost = (hostname) => {
+  return hostname === 'localhost' || hostname.endsWith('.localhost')
+}
+
+const { hostname, port, protocol } = location
+
+const localhost = isLocalHost(hostname) || isIp(hostname)
+
+const secure = protocol === 'https:'
+
+function getMimeType() {
+  if (!window.MediaRecorder) {
+    return
+  }
+
+  const types = [
+    'video/webm;codecs=h264',
+    'video/webm',
+    'video/webm;codecs=opus',
+    'video/webm;codecs=vp8',
+    'video/webm;codecs=daala',
+    'video/mpeg',
+    'video/mp4',
+  ]
+
+  let mimeType = ''
+  for (const type of types) {
+    const supported = MediaRecorder.isTypeSupported(type)
+    if (supported) {
+      mimeType = type
+      break
+    }
+  }
+
+  console.log('_mimeType', mimeType)
+  return mimeType
+}
+
+export const mimeType = getMimeType()
+
+function querystring(params: Record<string, any>) {
+  const escape = encodeURIComponent
+  const raw = Object.keys(params)
+    .filter((k) => !!params[k])
+    .map((k) => escape(k) + '=' + escape(params[k].toString()))
+    .join('&')
+  return raw.length === 0 ? '' : '?' + raw
+}
+
+function connect(streamKey: string, ignoreCookies: boolean) {
+  const protocol = !localhost && secure ? 'wss' : 'ws'
```

Comment:
In which case would we have `localhost && secure`?

Reply:
no idea, this was just copied from the code in `index`